File: Data Structures & Algorithms/is-palindrome/submission-3.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        word = s.strip().lower()
        word = "".join(c for c in word if c.isalnum())

        i ,j =0, len(word) - 1
        while i <= j:
            if word[i] != word[j]:
                return False
            i += 1
            j -= 1
        return True

File: Data Structures & Algorithms/valid-palindrome-ii/submission-3.java
class Solution {
    public boolean validPalindrome(String s) {
        return isPalindrome(s, 0, s.length() -1, 1);
    }

    public boolean isPalindrome(String s, int r, int l, int count){
        while (r < l){
            if (s.charAt(r) != s.charAt(l)){
                if (count == 0) return false;
                return isPalindrome(s, r+1, l, 0) || isPalindrome(s, r, l-1, 0);
            }
            r++;
            l--;
        }
        return true;
    }
}